app/models/donation.rb

```diff
@@ -44,4 +41,32 @@ def value
   def item_count
     donation_details.map(&:quantity).sum
   end
+
+  private
+
+  def skip_adding_donations?(params, required)
+    return false if required
+    return true if params[:donation].blank?
+    return true if params[:donation][:donation_details].blank?
+    return true if params[:donation][:donation_details][:item_id].blank?
```

Comment:
I think you can consolidate this logic here using `#dig`
```
params.dig(:donation, :donation_details, :item_id).blank?
```

Reply:
👍 +1